mlkem/indcpa.c

```diff
@@ -288,15 +349,35 @@ void indcpa_enc(uint8_t c[KYBER_INDCPA_BYTES],
     poly_frommsg(&k, m);
     gen_at(at, seed);
 
-    for (i = 0; i < KYBER_K; i++)
-    {
-        poly_getnoise_eta1(sp.vec + i, coins, nonce++);
-    }
-    for (i = 0; i < KYBER_K; i++)
-    {
-        poly_getnoise_eta2(ep.vec + i, coins, nonce++);
-    }
+    #if KYBER_K == 2
+    poly_getnoise_eta1(&sp.vec[0], coins, nonce++);
+    poly_getnoise_eta1(&sp.vec[1], coins, nonce++);
+
+    poly_getnoise_eta2(&ep.vec[0], coins, nonce++);
+    poly_getnoise_eta2(&ep.vec[1], coins, nonce++);
+
+    poly_getnoise_eta2(&epp, coins, nonce++);
+    #elif KYBER_K == 3
+    poly_getnoise_eta1(&sp.vec[0], coins, nonce++);
+    poly_getnoise_eta1(&sp.vec[1], coins, nonce++);
+    poly_getnoise_eta1(&sp.vec[2], coins, nonce++);
+
+    poly_getnoise_eta2(&ep.vec[0], coins, nonce++);
+    poly_getnoise_eta2(&ep.vec[1], coins, nonce++);
+    poly_getnoise_eta2(&ep.vec[2], coins, nonce++);
+
+    poly_getnoise_eta2(&epp, coins, nonce++);
+    #elif KYBER_K == 4
+    polyvec_getnoise_eta1(&sp, coins, nonce);
+    nonce += KYBER_K;
+
+    polyvec_getnoise_eta2(&ep, coins, nonce);
+    nonce += KYBER_K;
+
     poly_getnoise_eta2(&epp, coins, nonce++);
```

Comment:
The total of SHAKE256 calls here are 9. So we can have `KECCAK_WAY=4, 5` to efficiently compute keccak here.